ledger-core/v2/insolar/runner/runner.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package runner
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/payload"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/record"
+	"github.com/insolar/assured-ledger/ledger-core/v2/logicrunner/common"
+	"github.com/insolar/assured-ledger/ledger-core/v2/runner/requestresult"
+	"github.com/insolar/assured-ledger/ledger-core/v2/virtual/descriptor"
+)
+
+type ContractExecutionStateUpdateType int
+
+//go:generate stringer -type=ContractExecutionStateUpdateType
+
+const (
+	_ ContractExecutionStateUpdateType = iota
+
+	ContractError
+	ContractAborted
+	ContractDone
+
+	ContractOutgoingCall
+)
+
+type RPCEvent interface{}
+
+type ContractExecutionStateUpdate struct {
+	Type  ContractExecutionStateUpdateType
+	Error error
+
+	Result   *requestresult.RequestResult
+	Outgoing RPCEvent
+}
+
+type Execution struct {
+	Object     descriptor.ObjectDescriptor
+	Context    context.Context
+	Request    *payload.VCallRequest
+	Sequence   uint32
+	Deactivate bool
+	Pulse      insolar.Pulse
+
+	LogicContext insolar.LogicCallContext
+}
+
+type Runner interface {
```

Comment:
hmm, maybe runner.Service instead of runner.Runner, last one looks ugly, some linters would call this tautology